qcodes/dataset/descriptions.py

```diff
@@ -21,46 +21,47 @@ class RunDescriber:
     itself.
     """
 
-    def __init__(self, interdeps: SomeDeps) -> None:
+    def __init__(self, interdeps: InterDependencies_) -> None:
 
-        self._old_style_deps: bool
-
-        if isinstance(interdeps, InterDependencies_):
-            self._old_style_deps = False
-        elif isinstance(interdeps, InterDependencies):
-            self._old_style_deps = True
-        else:
+        if not isinstance(interdeps, InterDependencies_):
             raise ValueError('The interdeps arg must be of type: '
-                             'InterDependencies or InterDependencies_. '
-                             f' Got {type(interdeps)}.')
+                             'InterDependencies_. '
+                             f'Got {type(interdeps)}.')
 
         self.interdeps = interdeps
 
+        # we operate with two version numbers
+        # _version: the version of objects used inside this class
+        # _written_version: the version of objects we write to the DB
+        #
+        # We can not simply write the current version to disk, as some
+        # third-party applications may not handle that too well
+
+        self._version = 1
+        self._written_version = 0
```

Comment:
ah, and as always, a human-readable description of the versions is very much welcome :)